File: qc-application/src/main/java/com/eager/questioncloud/application/api/user/profile/UserProfileController.java
package com.eager.questioncloud.application.api.user.profile;

import com.eager.questioncloud.application.api.common.DefaultResponse;
import com.eager.questioncloud.application.api.user.profile.UserPointControllerRequest.UpdateMyInformationRequest;
import com.eager.questioncloud.application.api.user.profile.UserPointControllerResponse.MyInformationResponse;
import com.eager.questioncloud.application.security.UserPrincipal;
import com.eager.questioncloud.domain.user.MyInformation;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.security.core.annotation.AuthenticationPrincipal;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PatchMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/user/profile")
@RequiredArgsConstructor
public class UserProfileController {
    private final UserProfileService userService;

    @GetMapping("/me")
    public MyInformationResponse getMyInformation(@AuthenticationPrincipal UserPrincipal userPrincipal) {
        return new MyInformationResponse(MyInformation.from(userPrincipal.getUser()));
    }

    @PatchMapping("/me")
    public DefaultResponse updateMyInformation(
        @AuthenticationPrincipal UserPrincipal userPrincipal, @RequestBody @Valid UpdateMyInformationRequest request) {
        userService.updateUserInformation(userPrincipal.getUser(), request.getName(), request.getProfileImage());
        return DefaultResponse.success();
    }
}

File: qc-api/qc-user-api/src/main/kotlin/com/eager/questioncloud/user/profile/controller/UserProfileController.kt
package com.eager.questioncloud.user.profile.controller

import com.eager.questioncloud.common.auth.UserPrincipal
import com.eager.questioncloud.common.dto.DefaultResponse
import com.eager.questioncloud.user.profile.dto.MyInformationResponse
import com.eager.questioncloud.user.profile.dto.UpdateMyInformationRequest
import com.eager.questioncloud.user.profile.service.UserProfileService
import jakarta.validation.Valid
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/user/profile")
class UserProfileController(
    private val userProfileService: UserProfileService
) {
    @GetMapping("/me")
    fun getMyInformation(userPrincipal: UserPrincipal): MyInformationResponse {
        return MyInformationResponse(userProfileService.getMyInformation(userPrincipal.userId))
    }

    @PatchMapping("/me")
    fun updateMyInformation(
        userPrincipal: UserPrincipal, @RequestBody request: @Valid UpdateMyInformationRequest
    ): DefaultResponse {
        userProfileService.updateUserInformation(userPrincipal.userId, request.name, request.profileImage)
        return DefaultResponse.success()
    }
}
